android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.remotesettings
+
+import android.util.AtomicFile
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import mozilla.appservices.remotesettings.Attachment
+import mozilla.appservices.remotesettings.RemoteSettings
+import mozilla.appservices.remotesettings.RemoteSettingsConfig
+import mozilla.appservices.remotesettings.RemoteSettingsException
+import mozilla.appservices.remotesettings.RemoteSettingsRecord
+import mozilla.appservices.remotesettings.RsJsonObject
+import mozilla.components.support.base.log.logger.Logger
+import mozilla.components.support.ktx.util.writeString
+import org.json.JSONArray
+import org.json.JSONObject
+import java.io.File
+import java.io.FileReader
+
+/**
+ * Helper class to download collections from remote settings in app services.
+ *
+ * @property storageRootDirectory The top level app-local storage directory.
+ * @property serverUrl The optional url for the settings server. If not specified, the standard server will be used.
+ * @property bucketName The optional name of the bucket containing the collection on the server.
+ * If not specified, the standard bucket will be used.
+ * @property collectionName The name of the collection for the settings server.
+ */
+class RemoteSettingsClient(
+    private val storageRootDirectory: File,
+    val serverUrl: String = "https://firefox.settings.services.mozilla.com",
+    val bucketName: String = "main",
+    val collectionName: String = "",
+) {
+
+    private val config = RemoteSettingsConfig(
+        serverUrl = serverUrl,
+        bucketName = bucketName,
+        collectionName = collectionName,
+    )
+
+
+    /**
+     * Fetches all records for the collection from remote servers and updates the local storage.
+     **/
+    suspend fun fetchRemoteServerRecords(): List<RemoteSettingsRecord> = withContext(Dispatchers.IO) {
+        try {
+            val serverRecords = RemoteSettings(config).use {
+                it.getRecords().records
+            }
+            return@withContext serverRecords
+        } catch (e: RemoteSettingsException.ResponseException) {
+            emptyList()
+        } catch (e: Exception) {
+            Logger(e.message.toString())
+            emptyList()
+
+        }
+    }
+
+    private fun writeRemoteRecordsToStorage(file: File, records: List<RemoteSettingsRecord>) {
+        try {
+            val jsonArray = JSONArray()
+            for (record in records) {
```

Comment:
what you've got here is totally fine, I just wanted to mention you could also use the Kotlins Collection framework and do something like:
```
 records.forEach {
                val recordJson =
                    JSONObject().put("id", it.id)
...
}
```